src/app/Http/Controllers/API/PollController.php

```diff
@@ -105,7 +139,14 @@ public function destroy(Poll $poll)
      */
     public function publish(Request $request, Poll $poll)
     {
+        if (!PollController::checkPresenterPrivilege($request, $poll)) {
+            return response()->json('unauthorized', Response::HTTP_UNAUTHORIZED);
+        }
+
+        $poll->status = 'published';
```

Comment:
you should create an enum on Poll model for statuses

Reply:
Fixed as per last commit